### PR TITLE
Increase default write queue size

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit increases the default write queue size to 10000. This is to
allow a greater number of pending indexing requests. This work is safe
as we have added additional memory limits. Relates to #59263.

### CODE REVIEW

**Comment by tbrooks8** [comment]
After this is merged, it needs to be forward ported to 8.0 without the release notes. I have added a meta issue for that.

File: docs/reference/release-notes/7.9.asciidoc [hunk 1]
```diff
@@ -22,3 +22,10 @@ Snapshot restore throttling::
   default that's used for `indices.recovery.max_bytes_per_sec`. This means
   that no behavioral change will be observed by clusters where the recovery
   and restore settings had not been adapted from the defaults. {es-pull}58658[#58658]
+
+Thread pool write queue size::
+* The WRITE thread pool default queue size (`thread_pool.write.size`) has been
+  increased from 200 to 4000. Additional memory-oriented back pressure has been
+  introduced with the `indexing_limits.memory.limit` setting. This setting
```
Comment (ywelsch) [inline docs/reference/release-notes/7.9.asciidoc:29]: you mean `indexing_pressure.memory.limit` here?

**Review by ywelsch — CHANGES_REQUESTED** [review]
This is missing the actual change to the setting.

File: docs/reference/release-notes/7.9.asciidoc [hunk 2]
```diff
@@ -22,3 +22,10 @@ Snapshot restore throttling::
   default that's used for `indices.recovery.max_bytes_per_sec`. This means
   that no behavioral change will be observed by clusters where the recovery
   and restore settings had not been adapted from the defaults. {es-pull}58658[#58658]
+
+Thread pool write queue size::
+* The WRITE thread pool default queue size (`thread_pool.write.size`) has been
+  increased from 200 to 10000. Additional memory-oriented back pressure has been
```
Comment (ywelsch) [inline docs/reference/release-notes/7.9.asciidoc:28]: Should we mention here that this new model now better supports indexing loads that consist of a few large bulks as well as many small bulks (from a larger number of clients).

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by tbrooks8** [comment]
This is currently labelled breaking as it changes a major default setting - the `WRITE` queue size from 200->10,000.

**Pull request merged**